src/snapshot.js

```diff
@@ -93,6 +110,15 @@ export function restorePlayerSnapshot(player, snapshotObject) {
 
   const suppressedRemoteTracks = snapshotObject.suppressedRemoteTracks;
   const suppressedTracks = snapshotObject.suppressedTracks;
+
+  // ensures that the iOS TextTrackList 'change' listener added in getPlayerSnapshot()
+  // doesn't persist into main content
+  if (videojs.browser.IS_IOS) {
+    const tracks = player.textTracks();
+
+    tracks.off('change');
```

Comment:
Probably the easiest way is to still define the method inside of `getPlayerSnapshot()` but then keep a reference to it somewhere to be available in `restorePlayerSnapshot()`.